UI Freeze & Error: Cannot find module 'typescript'

Hi,

since the latest install (3 days before) I get these error messages in the log (via `journalctl -f`):
https://gist.github.com/jonashungershausen/e0baca010174924d47cb552b89e598b6
(Couldn't figure out how to properly format it here in the comment).

They go along with some 20 second complete UI freezes of the eclipse instance. 

Using: 
Spring Tool Suite 4 

Version: 4.0.0.RELEASE
Build Id: 201809220817

on Fedora 28

I also installed typescript globally via `yarn global add typescript` but that didn't seem to work.

Thanks for the report. I copy your gist here in case the gist disapeared. If you want to share some code/logs you can use 3 back thicks before and after the code block. 

```
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]: Error: Cannot find module 'typescript'
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Function.Module._resolveFilename (module.js:548:15)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Function.Module._load (module.js:475:25)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Module.require (module.js:597:17)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at require (internal/module.js:11:18)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Object.<anonymous> (/home/jhungershausen/Library/sts-4.0.0.RELEASE/plugins/org.eclipse.wildwebdeveloper_0.1.0.201810092241/language-servers/node_modules/vscode-html-languageserver/out/modes/javascriptMode.js:10:10)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Module._compile (module.js:653:30)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Object.Module._extensions..js (module.js:664:10)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at Module.load (module.js:566:32)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]:     at tryModuleLoad (module.js:506:12)
Oct 30 17:07:34 localhost.localdomain Spring Tool Suite 4.desktop[5431]: at Function.Module._load (module.js:498:3)
```
Do you know which version of node is include in your path?

Yes, I'd like to. Will do it this evening if that's okay. Do you have any pointers to resources I need to set up the development environment?

Outlook for Android<https://aka.ms/ghei36> herunterladen

________________________________
From: Mickael Istria <notifications@github.com>
Sent: Thursday, November 15, 2018 11:13:18 AM
To: eclipse/wildwebdeveloper
Cc: Jonas Hungershausen; Mention
Subject: Re: [eclipse/wildwebdeveloper] UI Freeze & Error: Cannot find module 'typescript' (#88)


@jonashungershausen<https://github.com/jonashungershausen> I tried your suggestion and it work very well. Would you like to submit the PR, as you're actually the one who made the report and found the best solution?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/eclipse/wildwebdeveloper/issues/88#issuecomment-438988451>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFSJn3E-rSuYHVve-MMcsMrT8jtOiNkTks5uvT4-gaJpZM4YCTy8>.

I agree with your last comment. I checked why it happens now and not in previous version of wild web developer. I think the replacement of `javascript-typescript-langserver` by `typescript-language-server` reveal the problem. 

Typescript is also declare as a dependency in `javascript-typescript-langserver` but it's not in `typescript-language-server`. 
I will provide an update of the package.json this evening. I think we should also update the target platform as the build of master failed.  

```
[ERROR] Cannot resolve target definition:
[ERROR]   Software being installed: org.eclipse.lsp4j.generator 0.6.0.v20181107-1212
[ERROR]   Missing requirement: org.eclipse.lsp4j.generator 0.6.0.v20181107-1212 requires 'java.package; org.eclipse.xtend.lib.annotations [2.16.0,3.0.0)' but it could not be found
[ERROR] 
```

--- a/org.eclipse.wildwebdeveloper/language-servers/package.json
+++ b/org.eclipse.wildwebdeveloper/language-servers/package.json
@@ -10,6 +10,7 @@
     "vscode-html-languageserver": "../target/vscode/VSCode-linux-x64/resources/app/extensions/html-language-features/server",
     "vscode-json-languageserver": "../target/vscode/VSCode-linux-x64/resources/app/extensions/json-language-features/server",
     "vscode-css-languageserver": "../target/vscode/VSCode-linux-x64/resources/app/extensions/css-language-features/server",
-    "typescript-language-server": "0.3.5"
+    "typescript-language-server": "0.3.5",
+    "typescript": "2.8.3"
   }
 }

